src/services/permissions.ts

```diff
@@ -0,0 +1,86 @@
+import { client } from "../banana-pan-chan";
+import config from "../config";
+import logger from "../utilities/logger";
+
+export enum Role {
+  Owner = 3,
+  Moderator = 2,
+  Vip = 1,
+  User = 0,
+}
+
+interface PermissionsMap {
+  [username: string]: Role;
+}
+
+export const AVAILABLE_PERMISSIONS = "(user|vips|mods|owner)";
```

Comment:
Perhaps generate this from `Role` rather than manually. If not, `user` should probably be `users` for consistency.